crates/core/src/network/connection_manager.rs

```diff
@@ -949,7 +949,7 @@ mod tests {
                 ProtocolError::HandshakeTimeout(_)
             ))
         ));
-        assert!(start.elapsed() >= net_opts.handshake_timeout.into());
+        assert!(start.elapsed() >= Into::<std::time::Duration>::into(net_opts.handshake_timeout));
```

Comment:
```suggestion
        assert!(start.elapsed() >= std::time::Duration::from(net_opts.handshake_timeout));
```

might be a bit simpler

Reply:
That was my first try, but this threw me off:

```
error[E0283]: type annotations needed
   --> crates/core/src/network/connection_manager.rs:952:89
    |
952 |         assert!(start.elapsed() >= std::time::Duration::from(net_opts.handshake_timeout.into()));
    |                                    ------------------- type must be known at this point ^^^^
    |
    = note: multiple `impl`s satisfying `std::time::Duration: derive_more::From<_>` found in the following crates: `aws_runtime`, `core`, `iso8601`:
            - impl derive_more::From<aws_runtime::service_clock_skew::ServiceClockSkew> for std::time::Duration;
            - impl derive_more::From<iso8601::duration::Duration> for std::time::Duration;
            - impl<T> derive_more::From<T> for T;
help: try using a fully qualified path to specify the expected types
    |
952 |         assert!(start.elapsed() >= std::time::Duration::from(<humantime::Duration as derive_more::Into<T>>::into(net_opts.handshake_timeout)));
    |                                                              ++++++++++++++++++++++++++++++++++++++++++++++++++++                          ~
```

There's a much simpler way though, updated! And thank you for the nudge in the right direction.